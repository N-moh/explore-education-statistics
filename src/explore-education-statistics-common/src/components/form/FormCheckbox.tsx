import classNames from 'classnames';
import React, { ChangeEventHandler } from 'react';
import styles from './FormCheckbox.module.scss';

export type CheckboxChangeEventHandler = ChangeEventHandler<HTMLInputElement>;

interface Props {
  checked?: boolean;
  id: string;
  hint?: string;
  label: string;
  name: string;
  onChange?: CheckboxChangeEventHandler;
  small?: boolean;
  value: string | string[] | number;
}

const FormCheckbox = ({
  checked,
  id,
  hint,
  label,
  name,
  onChange,
  small = false,
  value,
}: Props) => {
  return (
    <div
      className={classNames('govuk-checkboxes__item', {
        [styles.itemSmall]: small,
      })}
    >
      <input
        aria-describedby={hint ? `${id}-item-hint` : undefined}
        className="govuk-checkboxes__input"
        checked={checked}
        id={id}
        name={name}
        onChange={onChange}
        type="checkbox"
        value={value}
      />
      <label className="govuk-label govuk-checkboxes__label" htmlFor={id}>
        {label}
      </label>
      {hint && (
        <span
          id={`${id}-item-hint`}
          className="govuk-hint govuk-checkboxes__hint"
        >
          {hint}
        </span>
      )}
    </div>
  );
};

export default FormCheckbox;
