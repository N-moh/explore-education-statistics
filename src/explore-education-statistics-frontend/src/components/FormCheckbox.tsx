import React, { ChangeEventHandler, FunctionComponent } from 'react';

interface Props {
  checked?: boolean;
  id: string;
  hint?: string;
  label: string;
  name: string;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  value: string | string[] | number;
}

const FormCheckbox: FunctionComponent<Props> = ({
  checked,
  id,
  hint,
  label,
  name,
  onChange,
  value,
}) => {
  return (
    <div className="govuk-checkboxes__item">
      <input
        aria-describedby={hint ? `${id}-item-hint` : undefined}
        className="govuk-checkboxes__input"
        checked={checked}
        id={id}
        name={name}
        onChange={onChange}
        type="checkbox"
        value={value}
      />
      <label className="govuk-label govuk-checkboxes__label" htmlFor={id}>
        {label}
      </label>
      {hint && (
        <span
          id={`${id}-item-hint`}
          className="govuk-hint govuk-checkboxes__hint"
        >
          {hint}
        </span>
      )}
    </div>
  );
};

export default FormCheckbox;
