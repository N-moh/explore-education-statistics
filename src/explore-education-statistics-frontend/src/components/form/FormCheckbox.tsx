import React, { ChangeEventHandler, FunctionComponent } from 'react';

export type CheckboxChangeEventHandler<
  T = HTMLInputElement
> = ChangeEventHandler<T>;

interface Props {
  checked?: boolean;
  id: string;
  hint?: string;
  label: string;
  name: string;
  onChange?: CheckboxChangeEventHandler;
  value: string | string[] | number;
}

export const FormCheckbox: FunctionComponent<Props> = ({
  checked,
  id,
  hint,
  label,
  name,
  onChange,
  value,
}) => {
  return (
    <div className="govuk-checkboxes__item">
      <input
        aria-describedby={hint ? `${id}-item-hint` : undefined}
        className="govuk-checkboxes__input"
        checked={checked}
        id={id}
        name={name}
        onChange={onChange}
        type="checkbox"
        value={value}
      />
      <label className="govuk-label govuk-checkboxes__label" htmlFor={id}>
        {label}
      </label>
      {hint && (
        <span
          id={`${id}-item-hint`}
          className="govuk-hint govuk-checkboxes__hint"
        >
          {hint}
        </span>
      )}
    </div>
  );
};

export default FormCheckbox;
