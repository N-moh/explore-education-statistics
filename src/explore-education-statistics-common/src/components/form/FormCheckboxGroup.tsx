import { Omit } from '@common/types/util';
import classNames from 'classnames';
import React, { Component, createRef } from 'react';
import FormCheckbox, {
  CheckboxChangeEventHandler,
  FormCheckboxProps,
} from './FormCheckbox';
import FormFieldset, { FieldSetProps } from './FormFieldset';

export type CheckboxOption = Omit<
  FormCheckboxProps,
  'name' | 'checked' | 'onChange'
>;

export type FormCheckboxGroupProps = {
  name: string;
  onAllChange?: CheckboxChangeEventHandler;
  onChange?: CheckboxChangeEventHandler;
  options: CheckboxOption[];
  selectAll?: boolean;
  small?: boolean;
  value: string[];
} & FieldSetProps;

interface State {
  checkedCount: number;
}

/**
 * Basic checkbox group that should be used as a controlled component.
 * When using Formik, use {@see FormFieldRadioGroup} instead.
 */
class FormCheckboxGroup extends Component<FormCheckboxGroupProps, State> {
  public static defaultProps = {
    legendSize: 'm',
    selectAll: false,
    small: false,
    value: [],
  };

  private ref = createRef<HTMLDivElement>();

  public componentDidMount(): void {
    if (this.ref.current) {
      import('govuk-frontend/components/checkboxes/checkboxes').then(
        ({ default: GovUkCheckboxes }) => {
          new GovUkCheckboxes(this.ref.current).init();
        },
      );
    }
  }

  public render() {
    const {
      value,
      onAllChange,
      onChange,
      name,
      id,
      options,
      selectAll,
      small,
    } = this.props;
    const isAllChecked = options.every(
      option => value.indexOf(option.value) > -1,
    );

    return (
      <FormFieldset {...this.props}>
        <div
          className={classNames('govuk-checkboxes', {
            'govuk-checkboxes--small': small,
          })}
          ref={this.ref}
        >
          {options.length > 1 && selectAll && (
            <FormCheckbox
              id={`${id}-all`}
              label="Select all"
              name={name}
              value="select-all"
              checked={isAllChecked}
              onChange={event => {
                if (onAllChange) {
                  onAllChange(event);
                }
              }}
            />
          )}

          {options.map(option => (
            <FormCheckbox
              {...option}
              name={name}
              key={option.id}
              checked={value.indexOf(option.value) > -1}
              onChange={event => {
                if (onChange) {
                  onChange(event);
                }
              }}
            />
          ))}
        </div>
      </FormFieldset>
    );
  }
}

export default FormCheckboxGroup;
