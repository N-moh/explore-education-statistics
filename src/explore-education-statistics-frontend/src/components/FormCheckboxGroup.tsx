import Checkboxes from 'govuk-frontend/components/checkboxes/checkboxes';
import React, { Component, createRef } from 'react';
import FormCheckbox, { CheckboxChangeEventHandler } from './FormCheckbox';
import FormFieldSet, { FieldSetProps } from './FormFieldSet';

export interface CheckboxOption {
  checked?: boolean;
  hint?: string;
  id: string;
  label: string;
  value: string;
}

type Props = {
  checkedValues: {
    [value: string]: boolean;
  };
  name: string;
  onAllChange?: CheckboxChangeEventHandler;
  onChange?: CheckboxChangeEventHandler<any>;
  options: CheckboxOption[];
} & Partial<FieldSetProps>;

interface State {
  checkedCount: number;
}

class FormCheckboxGroup extends Component<Props, State> {
  public static defaultProps: Partial<Props> = {
    checkedValues: {},
    legendSize: 'm',
  };

  private ref = createRef<HTMLDivElement>();

  public componentDidMount(): void {
    new Checkboxes(this.ref.current).init();
  }

  private handleChange: CheckboxChangeEventHandler = event => {
    if (this.props.onChange) {
      this.props.onChange(event);
    }
  };

  private handleAllChange: CheckboxChangeEventHandler = event => {
    if (this.props.onAllChange) {
      this.props.onAllChange(event);
    }
  };

  private renderCheckboxes() {
    const { checkedValues, onAllChange, name, options } = this.props;

    const checkedCount = Object.values(checkedValues).filter(Boolean).length;

    return (
      <div className="govuk-checkboxes" ref={this.ref}>
        {onAllChange && (
          <FormCheckbox
            id={`${name}-all`}
            label="Select all"
            name={name}
            value="select-all"
            checked={checkedCount === options.length}
            onChange={this.handleAllChange}
          />
        )}

        {options.map(option => (
          <FormCheckbox
            {...option}
            name={name}
            key={option.id}
            onChange={this.handleChange}
            checked={Boolean(checkedValues[option.value])}
          />
        ))}
      </div>
    );
  }

  public render() {
    const { legend, ...restProps } = this.props;

    return legend ? (
      <FormFieldSet {...restProps} legend={legend}>
        {this.renderCheckboxes()}
      </FormFieldSet>
    ) : (
      this.renderCheckboxes()
    );
  }
}

export default FormCheckboxGroup;
