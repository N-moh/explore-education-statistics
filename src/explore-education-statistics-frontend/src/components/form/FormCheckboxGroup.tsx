import Checkboxes from 'govuk-frontend/components/checkboxes/checkboxes';
import React, { Component, createRef } from 'react';
import FormCheckbox, { CheckboxChangeEventHandler } from './FormCheckbox';
import FormFieldSet, { FieldSetProps } from './FormFieldSet';

export interface CheckboxOption {
  hint?: string;
  id: string;
  label: string;
  value: string;
}

type Props = {
  name: string;
  onAllChange?: CheckboxChangeEventHandler;
  onChange?: CheckboxChangeEventHandler<any>;
  options: CheckboxOption[];
  value: string[];
} & FieldSetProps;

interface State {
  checkedCount: number;
}

class FormCheckboxGroup extends Component<Props, State> {
  public static defaultProps: Partial<Props> = {
    legendSize: 'm',
    value: [],
  };

  private ref = createRef<HTMLDivElement>();

  public componentDidMount(): void {
    new Checkboxes(this.ref.current).init();
  }

  public render() {
    const { value, onAllChange, name, options } = this.props;

    const checkedCount = Object.values(value).filter(Boolean).length;

    return (
      <FormFieldSet {...this.props}>
        <div className="govuk-checkboxes" ref={this.ref}>
          {onAllChange && (
            <FormCheckbox
              id={`${name}-all`}
              label="Select all"
              name={name}
              value="select-all"
              checked={checkedCount === options.length}
              onChange={event => {
                if (this.props.onAllChange) {
                  this.props.onAllChange(event);
                }
              }}
            />
          )}

          {options.map(option => (
            <FormCheckbox
              {...option}
              name={name}
              key={option.id}
              checked={value.indexOf(option.value) > -1}
              onChange={event => {
                if (this.props.onChange) {
                  this.props.onChange(event);
                }
              }}
            />
          ))}
        </div>
      </FormFieldSet>
    );
  }
}

export default FormCheckboxGroup;
