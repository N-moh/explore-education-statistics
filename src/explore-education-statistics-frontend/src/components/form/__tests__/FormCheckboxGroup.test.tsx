import React, { Component, ReactNode } from 'react';
import { fireEvent, render } from 'react-testing-library';
import { CheckboxChangeEventHandler } from '../FormCheckbox';
import FormCheckboxGroup from '../FormCheckboxGroup';

describe('FormCheckboxGroup', () => {
  // Stateful container component to wrap our checkbox groups with
  class CheckboxWrapper extends Component<
    {
      children: (
        state: string[],
        handleChange: CheckboxChangeEventHandler,
      ) => ReactNode;
      initialState: string[];
    },
    {
      values: string[];
    }
  > {
    public state = {
      values: [...this.props.initialState],
    };

    private handleChange: CheckboxChangeEventHandler = event => {
      const values: string[] = [...this.state.values];

      if (event.target.checked) {
        values.push(event.target.value);
      } else {
        values.splice(values.indexOf(event.target.value), 1);
      }

      this.setState({ values });
    };

    public render() {
      return this.props.children(this.state.values, this.handleChange);
    }
  }

  test('renders list of checkboxes in correct order', () => {
    const { container, getAllByLabelText } = render(
      <FormCheckboxGroup
        id="test-checkboxes"
        name="test-checkboxes"
        options={[
          { id: 'checkbox-1', label: 'Test checkbox 1', value: '1' },
          { id: 'checkbox-2', label: 'Test checkbox 2', value: '2' },
          { id: 'checkbox-3', label: 'Test checkbox 3', value: '3' },
        ]}
      />,
    );

    const checkboxes = getAllByLabelText(/Test checkbox/);

    expect(checkboxes).toHaveLength(3);
    expect(checkboxes[0]).toHaveAttribute('value', '1');
    expect(checkboxes[1]).toHaveAttribute('value', '2');
    expect(checkboxes[2]).toHaveAttribute('value', '3');

    expect(container.innerHTML).toMatchSnapshot();
  });

  test('renders checkboxes with some pre-checked', () => {
    const { container, getAllByLabelText } = render(
      <FormCheckboxGroup
        id="test-checkboxes"
        name="test-checkboxes"
        options={[
          { id: 'checkbox-1', label: 'Test checkbox 1', value: '1' },
          { id: 'checkbox-2', label: 'Test checkbox 2', value: '2' },
          { id: 'checkbox-3', label: 'Test checkbox 3', value: '3' },
        ]}
        value={['2']}
      />,
    );

    const checkboxes = getAllByLabelText(/Test checkbox/) as HTMLInputElement[];

    expect(checkboxes).toHaveLength(3);
    expect(checkboxes[0].checked).toBe(false);
    expect(checkboxes[1].checked).toBe(true);
    expect(checkboxes[2].checked).toBe(false);

    expect(container.innerHTML).toMatchSnapshot();
  });

  test('clicking an unchecked checkbox checks it', () => {
    const { getByLabelText } = render(
      <CheckboxWrapper initialState={[]}>
        {(state, handleChange) => (
          <FormCheckboxGroup
            value={state}
            onChange={handleChange}
            options={[{ id: 'checkbox-1', label: 'Test checkbox', value: '1' }]}
            id="test-checkboxes"
            name="test-checkboxes"
          />
        )}
      </CheckboxWrapper>,
    );

    const checkbox = getByLabelText('Test checkbox') as HTMLInputElement;

    expect(checkbox.checked).toBe(false);

    fireEvent.click(checkbox);

    expect(checkbox.checked).toBe(true);
  });

  test('clicking a pre-checked checkbox un-checks it', () => {
    const { getByLabelText } = render(
      <CheckboxWrapper initialState={['1']}>
        {(state, handleChange) => (
          <FormCheckboxGroup
            value={state}
            onChange={handleChange}
            options={[{ id: 'checkbox-1', label: 'Test checkbox', value: '1' }]}
            id="test-checkboxes"
            name="test-checkboxes"
          />
        )}
      </CheckboxWrapper>,
    );

    const checkbox = getByLabelText('Test checkbox') as HTMLInputElement;

    expect(checkbox.checked).toBe(true);

    fireEvent.click(checkbox);

    expect(checkbox.checked).toBe(false);
  });

  test('clicking multiple checkboxes checks them all', () => {
    const { getByLabelText } = render(
      <CheckboxWrapper initialState={[]}>
        {(state, handleChange) => (
          <FormCheckboxGroup
            value={state}
            id="test-checkboxes"
            name="test-checkboxes"
            onChange={handleChange}
            options={[
              { id: 'checkbox-1', label: 'Test checkbox 1', value: '1' },
              { id: 'checkbox-2', label: 'Test checkbox 2', value: '2' },
              { id: 'checkbox-3', label: 'Test checkbox 3', value: '3' },
            ]}
          />
        )}
      </CheckboxWrapper>,
    );

    const checkbox1 = getByLabelText('Test checkbox 1') as HTMLInputElement;
    const checkbox2 = getByLabelText('Test checkbox 2') as HTMLInputElement;
    const checkbox3 = getByLabelText('Test checkbox 3') as HTMLInputElement;

    expect(checkbox1.checked).toBe(false);
    expect(checkbox2.checked).toBe(false);
    expect(checkbox3.checked).toBe(false);

    fireEvent.click(checkbox1);
    fireEvent.click(checkbox2);

    expect(checkbox1.checked).toBe(true);
    expect(checkbox2.checked).toBe(true);
    expect(checkbox3.checked).toBe(false);
  });

  test('renders correctly with legend', () => {
    const { container, getByText } = render(
      <FormCheckboxGroup
        legend="Choose some checkboxes"
        id="test-checkboxes"
        name="test-checkboxes"
        options={[
          { id: 'radio-1', label: 'Test radio 1', value: '1' },
          { id: 'radio-2', label: 'Test radio 2', value: '2' },
          { id: 'radio-3', label: 'Test radio 3', value: '3' },
        ]}
      />,
    );

    expect(getByText('Choose some checkboxes')).toBeDefined();
    expect(container.innerHTML).toMatchSnapshot();
  });

  test('renders with `Select all` checkbox when `onAllChange` is set', () => {
    const noop = () => null;

    const { container, getByLabelText } = render(
      <FormCheckboxGroup
        value={[]}
        id="test-checkboxes"
        name="test-checkboxes"
        onAllChange={noop}
        options={[
          { id: 'checkbox-1', label: 'Test checkbox 1', value: '1' },
          { id: 'checkbox-2', label: 'Test checkbox 2', value: '2' },
          { id: 'checkbox-3', label: 'Test checkbox 3', value: '3' },
        ]}
      />,
    );

    expect(getByLabelText('Select all')).toBeDefined();
    expect(container.innerHTML).toMatchSnapshot();
  });

  test('checking all options checks the `Select all` checkbox', () => {
    const noop = () => null;

    const { getByLabelText } = render(
      <CheckboxWrapper initialState={['2', '3']}>
        {(state, handleChange) => (
          <FormCheckboxGroup
            value={state}
            id="test-checkboxes"
            name="test-checkboxes"
            onAllChange={noop}
            onChange={handleChange}
            options={[
              { id: 'checkbox-1', label: 'Test checkbox 1', value: '1' },
              { id: 'checkbox-2', label: 'Test checkbox 2', value: '2' },
              { id: 'checkbox-3', label: 'Test checkbox 3', value: '3' },
            ]}
          />
        )}
      </CheckboxWrapper>,
    );

    const checkbox1 = getByLabelText('Test checkbox 1') as HTMLInputElement;
    const selectAll = getByLabelText('Select all') as HTMLInputElement;

    expect(checkbox1.checked).toBe(false);
    expect(selectAll.checked).toBe(false);

    fireEvent.click(checkbox1);

    expect(checkbox1.checked).toBe(true);
    expect(selectAll.checked).toBe(true);
  });

  test('un-checking any options un-checks the `Select all` checkbox ', () => {
    const noop = () => null;

    const { getByLabelText } = render(
      <CheckboxWrapper initialState={['1', '2', '3']}>
        {(state, handleChange) => (
          <FormCheckboxGroup
            value={state}
            id="test-checkboxes"
            name="test-checkboxes"
            onAllChange={noop}
            onChange={handleChange}
            options={[
              { id: 'checkbox-1', label: 'Test checkbox 1', value: '1' },
              { id: 'checkbox-2', label: 'Test checkbox 2', value: '2' },
              { id: 'checkbox-3', label: 'Test checkbox 3', value: '3' },
            ]}
          />
        )}
      </CheckboxWrapper>,
    );

    const checkbox1 = getByLabelText('Test checkbox 1') as HTMLInputElement;
    const selectAll = getByLabelText('Select all') as HTMLInputElement;

    expect(checkbox1.checked).toBe(true);
    expect(selectAll.checked).toBe(true);

    fireEvent.click(checkbox1);

    expect(checkbox1.checked).toBe(false);
    expect(selectAll.checked).toBe(false);
  });
});
