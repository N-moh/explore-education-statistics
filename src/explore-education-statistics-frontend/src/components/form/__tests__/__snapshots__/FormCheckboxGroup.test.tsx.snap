// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCheckboxGroup renders checkboxes with some pre-checked 1`] = `

<div class="govuk-form-group">
  <fieldset class="govuk-fieldset"
            id="test-checkboxes"
  >
    <div class="govuk-checkboxes">
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="checkbox-1"
               name="test-checkboxes"
               type="checkbox"
               value="1"
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="checkbox-1"
        >
          Test checkbox 1
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="checkbox-2"
               name="test-checkboxes"
               type="checkbox"
               value="2"
               checked
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="checkbox-2"
        >
          Test checkbox 2
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="checkbox-3"
               name="test-checkboxes"
               type="checkbox"
               value="3"
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="checkbox-3"
        >
          Test checkbox 3
        </label>
      </div>
    </div>
  </fieldset>
</div>

`;

exports[`FormCheckboxGroup renders correctly with legend 1`] = `

<div class="govuk-form-group">
  <fieldset class="govuk-fieldset"
            id="test-checkboxes"
  >
    <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">
      Choose some checkboxes
    </legend>
    <div class="govuk-checkboxes">
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="radio-1"
               name="test-checkboxes"
               type="checkbox"
               value="1"
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="radio-1"
        >
          Test radio 1
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="radio-2"
               name="test-checkboxes"
               type="checkbox"
               value="2"
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="radio-2"
        >
          Test radio 2
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="radio-3"
               name="test-checkboxes"
               type="checkbox"
               value="3"
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="radio-3"
        >
          Test radio 3
        </label>
      </div>
    </div>
  </fieldset>
</div>

`;

exports[`FormCheckboxGroup renders list of checkboxes in correct order 1`] = `

<div class="govuk-form-group">
  <fieldset class="govuk-fieldset"
            id="test-checkboxes"
  >
    <div class="govuk-checkboxes">
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="checkbox-1"
               name="test-checkboxes"
               type="checkbox"
               value="1"
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="checkbox-1"
        >
          Test checkbox 1
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="checkbox-2"
               name="test-checkboxes"
               type="checkbox"
               value="2"
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="checkbox-2"
        >
          Test checkbox 2
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="checkbox-3"
               name="test-checkboxes"
               type="checkbox"
               value="3"
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="checkbox-3"
        >
          Test checkbox 3
        </label>
      </div>
    </div>
  </fieldset>
</div>

`;

exports[`FormCheckboxGroup renders unchecked \`Select all\` checkbox when \`onAllChange\` is set 1`] = `

<div class="govuk-form-group">
  <fieldset class="govuk-fieldset"
            id="test-checkboxes"
  >
    <div class="govuk-checkboxes">
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="test-checkboxes-all"
               name="test-checkboxes"
               type="checkbox"
               value="select-all"
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="test-checkboxes-all"
        >
          Select all
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="checkbox-1"
               name="test-checkboxes"
               type="checkbox"
               value="1"
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="checkbox-1"
        >
          Test checkbox 1
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="checkbox-2"
               name="test-checkboxes"
               type="checkbox"
               value="2"
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="checkbox-2"
        >
          Test checkbox 2
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input"
               id="checkbox-3"
               name="test-checkboxes"
               type="checkbox"
               value="3"
        >
        <label class="govuk-label govuk-checkboxes__label"
               for="checkbox-3"
        >
          Test checkbox 3
        </label>
      </div>
    </div>
  </fieldset>
</div>

`;
